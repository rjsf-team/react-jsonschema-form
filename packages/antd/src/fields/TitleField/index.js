import React from 'react';
import classNames from 'classnames';

import { withConfigConsumer } from 'antd/lib/config-provider/context';

const TitleField = ({
  // autofocus,
  // disabled,
  // errorSchema,
  formContext,
  // formData,
  id,
  // idSchema,
  // name,
  // onChange,
  prefixCls,
  // readonly,
  // registry,
  required,
  // schema,
  title,
  // uiSchema,
}) => {
  const { colon = true } = formContext;

  let labelChildren = title;
  if (colon && typeof title === 'string' && title.trim() !== '') {
    labelChildren = title.replace(/[ï¼š:]\s*$/, '');
  }

  const labelClassName = classNames({
    [`${prefixCls}-item-required`]: required,
    [`${prefixCls}-item-no-colon`]: !colon,
  });

  const handleLabelClick = () => {
    if (!id) {
      return;
    }

    const control = document.querySelector(`[id="${id}"]`);
    if (control && control.focus) {
      control.focus();
    }
  };

  return title ? (
    <label
      className={labelClassName}
      htmlFor={id}
      onClick={handleLabelClick}
      title={typeof title === 'string' ? title : ''}
    >
      {labelChildren}
    </label>
  ) : null;
};

TitleField.defaultProps = {
  formContext: {},
};

export default withConfigConsumer({ prefixCls: 'form' })(TitleField);
