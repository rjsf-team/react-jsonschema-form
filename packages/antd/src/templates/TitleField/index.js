import React from "react";
import classNames from "classnames";

import { withConfigConsumer } from "antd/lib/config-provider/context";

const TitleField = ({ id, prefixCls, required, registry, title }) => {
  const { formContext } = registry;
  const { colon = true } = formContext;

  let labelChildren = title;
  if (colon && typeof title === "string" && title.trim() !== "") {
    labelChildren = title.replace(/[ï¼š:]\s*$/, "");
  }

  const labelClassName = classNames({
    [`${prefixCls}-item-required`]: required,
    [`${prefixCls}-item-no-colon`]: !colon,
  });

  const handleLabelClick = () => {
    if (!id) {
      return;
    }

    const control = document.querySelector(`[id="${id}"]`);
    if (control && control.focus) {
      control.focus();
    }
  };

  return title ? (
    <label
      className={labelClassName}
      htmlFor={id}
      onClick={handleLabelClick}
      title={typeof title === "string" ? title : ""}
    >
      {labelChildren}
    </label>
  ) : null;
};

TitleField.defaultProps = {
  formContext: {},
};

export default withConfigConsumer({ prefixCls: "form" })(TitleField);
