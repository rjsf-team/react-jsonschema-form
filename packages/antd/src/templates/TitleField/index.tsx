import classNames from 'classnames';
import { ConfigConsumer, ConfigConsumerProps } from 'antd/lib/config-provider/context';
import { FormContextType, TitleFieldProps, RJSFSchema, StrictRJSFSchema } from '@rjsf/utils';

/** The `TitleField` is the template to use to render the title of a field
 *
 * @param props - The `TitleFieldProps` for this component
 */
export default function TitleField<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>({
  id,
  required,
  registry,
  title,
}: TitleFieldProps<T, S, F>) {
  const { formContext } = registry;
  const { colon = true } = formContext;

  let labelChildren = title;
  if (colon && typeof title === 'string' && title.trim() !== '') {
    labelChildren = title.replace(/[ï¼š:]\s*$/, '');
  }

  const handleLabelClick = () => {
    if (!id) {
      return;
    }

    const control: HTMLLabelElement | null = document.querySelector(`[id="${id}"]`);
    if (control && control.focus) {
      control.focus();
    }
  };

  return title ? (
    <ConfigConsumer>
      {(configProps: ConfigConsumerProps) => {
        const { getPrefixCls } = configProps;
        const prefixCls = getPrefixCls('form');
        const labelClassName = classNames({
          [`${prefixCls}-item-required`]: required,
          [`${prefixCls}-item-no-colon`]: !colon,
        });

        return (
          <label
            className={labelClassName}
            htmlFor={id}
            onClick={handleLabelClick}
            title={typeof title === 'string' ? title : ''}
          >
            {labelChildren}
          </label>
        );
      }}
    </ConfigConsumer>
  ) : null;
}
