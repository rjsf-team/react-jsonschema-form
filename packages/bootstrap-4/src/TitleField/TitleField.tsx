import React from "react";
import { FieldProps } from "@rjsf/core";

export interface TitleFieldProps extends Partial<FieldProps> {
  title: string;
}

const TitleField = ({ title }: Partial<FieldProps>) => (
  <>
    <div className="my-1">
      <h5>{title}</h5>
      <hr className="border-0 bg-secondary" style={{ height: "1px" }} />
    </div>
  </>
);

export default TitleField;
