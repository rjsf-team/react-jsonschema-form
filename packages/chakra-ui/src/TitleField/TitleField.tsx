import { FormContextType, RJSFSchema, StrictRJSFSchema, TitleFieldProps } from '@rjsf/utils';
import { Box, Divider, Heading } from '@chakra-ui/react';

export default function TitleField<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>({
  id,
  title,
}: TitleFieldProps<T, S, F>) {
  return (
    <Box id={id} mt={1} mb={4}>
      <Heading as='h5'>{title}</Heading>
      <Divider />
    </Box>
  );
}
