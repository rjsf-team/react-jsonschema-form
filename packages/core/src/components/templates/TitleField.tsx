import { FormContextType, TitleFieldProps, RJSFSchema, StrictRJSFSchema } from '@rjsf/utils';

const REQUIRED_FIELD_SYMBOL = '*';

/** The `TitleField` is the template to use to render the title of a field
 *
 * @param props - The `TitleFieldProps` for this component
 */
export default function TitleField<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>(
  props: TitleFieldProps<T, S, F>
) {
  const { id, title, required } = props;
  return (
    <legend id={id}>
      {title}
      {required && <span className='required'>{REQUIRED_FIELD_SYMBOL}</span>}
    </legend>
  );
}
