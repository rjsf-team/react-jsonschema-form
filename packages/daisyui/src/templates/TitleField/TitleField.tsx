import { TitleFieldProps, StrictRJSFSchema, RJSFSchema, FormContextType, getUiOptions } from '@rjsf/utils';

/** The `TitleField` component renders the title for a form section or field
 * with DaisyUI styling. It displays:
 *
 * - Large heading with primary color
 * - Divider element below the title for visual separation
 * - Support for title override from UI options
 *
 * This component is typically used at the top of form sections to provide clear visual hierarchy.
 *
 * @param props - The `TitleFieldProps` for the component
 */
export default function TitleField<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>(
  props: TitleFieldProps<T, S, F>,
) {
  const { id, title, uiSchema } = props;
  const uiOptions = getUiOptions<T, S, F>(uiSchema);

  return (
    <div id={id} className='title-field mb-6'>
      <h2 className='text-3xl font-bold text-primary mb-2'>{uiOptions.title || title}</h2>
      <div className='divider divider-primary'></div>
    </div>
  );
}
