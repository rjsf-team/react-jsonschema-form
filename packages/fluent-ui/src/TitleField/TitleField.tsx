import React from "react";
import { FieldProps } from "@rjsf/core";
import { Label } from "@fluentui/react";

const styles = {
  root: [
    {
      fontSize: 24,
      marginBottom: 20,
      paddingBottom: 0
    },
  ],
};

const TitleField = ({ title }: FieldProps) => (
  <>
    <Label styles={styles}>{title}</Label>
  </>
);

export default TitleField;
