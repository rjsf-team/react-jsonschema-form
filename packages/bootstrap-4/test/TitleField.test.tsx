import React from "react";
import TitleField from "../src/TitleField";
import renderer from "react-test-renderer";

describe("TitleField", () => {
  test("simple", () => {
    const tree = renderer.create(<TitleField title="Hello" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
