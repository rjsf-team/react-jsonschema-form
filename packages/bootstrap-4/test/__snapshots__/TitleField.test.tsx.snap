// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleField simple 1`] = `
<div
  className="my-1"
>
  <h5>
    Hello
  </h5>
  <hr
    className="border-0 bg-secondary"
    style={
      Object {
        "height": "1px",
      }
    }
  />
</div>
`;
